exp/runtime/hooks/api/v1alpha1/common_types.go

```diff
@@ -20,6 +20,31 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// RequestObject is a runtime.Object extended with methods to handle request-specific fields.
+// +kubebuilder:object:generate=false
+type RequestObject interface {
+	runtime.Object
+	GetSettings() map[string]string
+	SetSettings(settings map[string]string)
+}
+
+// CommonRequest is the data structure common to all request types.
+// Note: By embedding CommonRequest in a runtime.Object the RequestObject
+// interface is satisfied.
+type CommonRequest struct {
```

Comment:
just noticing now that we have a small inconsistency between `CommonRequest`, `CommonResponse`, `ResponseStatus`, `CommonRetryResponse`, and `RetryResponseObject` (which has an Object suffix and doesn't have a Common prefix). Wondering if we should/could fix this in a follow-up PR
@killianmuldoon @sbueringer opinions?

Reply:
This was intentional. Our interfaces are called ResponseObject and RetryResponseObject (infered roughly from runtime.Object which is embedded).

Then we have CommonResponse and CommonRetryResponse which are structs that are shared across multiple response structs

Reply:
So I would keep it as is.

Changing it should also bring up the question of if we have to bump to v1alpha2

Reply:
+1 to the current naming scheme.